src/crimson/os/seastore/onode_manager/staged-fltree/stages/key_layout.h

```diff
@@ -555,6 +556,11 @@ inline std::ostream& operator<<(std::ostream& os, const key_hobj_t& key) {
  */
 class key_view_t {
  public:
+  static constexpr int MAX_NS_OID_LENGTH =
+    (4096 - sizeof(onode_layout_t) * 2) / 4
+    - sizeof(shard_pool_t) - sizeof(crush_t) - sizeof(snap_gen_t)
+    - 8; // size of length field of oid and ns
```

Comment:
To be precise this should include the overhead introduced by internal indexing and node/stage/value headers.

But I'm fine with this because actually `MAX_NS_OID_LENGTH` is defined by OSD and we will need to adjust the node size accordingly.

Maybe it is appropriate to add a `// FIXME` here?

Reply:
@cyx1231st Do you mean that we drop this change and add a FIXME for a change?

Reply:
No, add a FIXME to remember that `MAX_NS_OID_LENGTH` should be defined by OSD in the future.

Reply:
ok